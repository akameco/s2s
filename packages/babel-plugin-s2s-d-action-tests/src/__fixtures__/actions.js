// @flow
export type Action = Increment | FetchUser
