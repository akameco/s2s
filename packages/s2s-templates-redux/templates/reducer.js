// @flow
import type { Action } from '../../types'
import { Actions } from './actionTypes'

export type State = {}

export const initialState: State = {}

export default function(
  state: State = initialState,
  action: Action
): Exact<State> {
  switch (action.type) {
    default:
      return state
  }
}
