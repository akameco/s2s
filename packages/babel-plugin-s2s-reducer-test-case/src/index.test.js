// @flow
import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

const getFixturesPath = x => path.resolve(__dirname, '__fixtures__', x)
const fromPath = getFixturesPath('reducer.js')
const fromWithStatePath = getFixturesPath('with-initial-state.js')

const testCases = [
  {
    title: 'basic',
    code: `
import reducer, {initialState} from './actions'
    `,
  },
  {
    title: 'exist test case',
    code: `
import reducer, {initialState} from './actions'
import * as actions from './actions'

test('handle INCREMENT', () => {
  expect(actions.sample()).toEqual(null)
})
    `,
  },
]

pluginTester({
  title: 'default',
  plugin,
  snapshot: true,
  babelOptions: { configFile: false },
  pluginOptions: { from: fromPath },
  tests: testCases,
})

pluginTester({
  title: 'with initial state',
  plugin,
  snapshot: true,
  babelOptions: { configFile: false },
  pluginOptions: { from: fromWithStatePath },
  tests: testCases,
})

pluginTester({
  plugin,
  babelOptions: { configFile: false },
  tests: [
    {
      title: 'throw error',
      code: `// throw error`,
      error: /required from option/,
    },
  ],
})
