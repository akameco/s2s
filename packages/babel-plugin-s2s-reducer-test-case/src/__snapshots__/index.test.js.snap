// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default basic: basic 1`] = `
"
import reducer, {initialState} from './actions'

      ↓ ↓ ↓ ↓ ↓ ↓

import reducer, { initialState } from './actions';

test('handle INCREMENT', () => {
  expect(reducer(initialState, actions.increment())).toEqual(null);
});

test('handle DECREMENT', () => {
  expect(reducer(initialState, actions.decrement())).toEqual(null);
});
"
`;

exports[`default exist test case: exist test case 1`] = `
"
import reducer, {initialState} from './actions'
import * as actions from './actions'

test('handle INCREMENT', () => {
  expect(actions.sample()).toEqual(null)
})

      ↓ ↓ ↓ ↓ ↓ ↓

import reducer, { initialState } from './actions';
import * as actions from './actions';

test('handle INCREMENT', () => {
  expect(actions.sample()).toEqual(null);
});

test('handle DECREMENT', () => {
  expect(reducer(initialState, actions.decrement())).toEqual(null);
});
"
`;

exports[`with initial state basic: basic 1`] = `
"
import reducer, {initialState} from './actions'

      ↓ ↓ ↓ ↓ ↓ ↓

import reducer, { initialState } from './actions';

test('handle INCREMENT', () => {
  expect(reducer(initialState, actions.increment())).toEqual({ count: 0 });
});

test('handle DECREMENT', () => {
  expect(reducer(initialState, actions.decrement())).toEqual({ count: 0 });
});
"
`;

exports[`with initial state exist test case: exist test case 1`] = `
"
import reducer, {initialState} from './actions'
import * as actions from './actions'

test('handle INCREMENT', () => {
  expect(actions.sample()).toEqual(null)
})

      ↓ ↓ ↓ ↓ ↓ ↓

import reducer, { initialState } from './actions';
import * as actions from './actions';

test('handle INCREMENT', () => {
  expect(actions.sample()).toEqual(null);
});

test('handle DECREMENT', () => {
  expect(reducer(initialState, actions.decrement())).toEqual({ count: 0 });
});
"
`;
